### PR TITLE
Create ES index templates instead of indices

### LINKED ISSUE #374: ElasticSearch index can be created with incorrect mapping
When collector was running I've changed the ES cluster. As a result an index was created with incorrect mapping - 
```
{
    "jaeger-span-2017-09-03": {
        "mappings": {
            "span": {
                "properties": {

                    (removed for brevity)

                    "traceID": {
                        "fields": {
                            "keyword": {
                                "ignore_above": 256,
                                "type": "keyword"
                            }
                        },
                        "type": "text"
                    }
                }
            }
        }
    }
}
```

I suspect that setting `dynamic=false` on the top-level tocument or `action.auto_create_index: false` would solve the problem.

### LINKED ISSUE #622: query error after run es_indices_clean.sh
jaeger version: cba413ede300564e6f92a16f812604e4a829d0c7
storage: elasticsearch 6.1.1, Build: bd92e7f/2017-12-17T20:23:25.338Z, JVM: 9.0.1

It's work before clean.

My operation process:
```
⋊> ~/opt ./es_indices_clean.sh 0 localhost:9200                                                                                                                                                             
Installing python dependencies required for curator...
Requirement already satisfied: elasticsearch in /usr/local/lib/python2.7/dist-packages
Requirement already satisfied: elasticsearch-curator in /usr/local/lib/python2.7/dist-packages
Requirement already satisfied: urllib3<1.23,>=1.21.1 in /usr/local/lib/python2.7/dist-packages (from elasticsearch)
Requirement already satisfied: click>=6.7 in /usr/local/lib/python2.7/dist-packages (from elasticsearch-curator)
Requirement already satisfied: pyyaml>=3.10 in /usr/local/lib/python2.7/dist-packages (from elasticsearch-curator)
Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python2.7/dist-packages (from elasticsearch-curator)
Requirement already satisfied: voluptuous>=0.9.3 in /usr/local/lib/python2.7/dist-packages (from elasticsearch-curator)

Removing jaeger-service-2017-12-26
Removing jaeger-span-2017-12-26
⋊> ~/opt curl -XGET 'http://localhost:9200/_cat/indices'          # generate new data                                                                                                                                          
yellow open jaeger-service-2017-12-26 sMNJrlcYQGqMDp4AUrLlWA 5 1  2 0   9.9kb   9.9kb
yellow open jaeger-span-2017-12-26    wBc1qghHQ4uo0XobS7tKgA 5 1 27 0 263.7kb 263.7kb
```

 Then visit `http://local:16686/search`, get error:
```
There was an error querying for traces:
HTTP Error: Search service failed: elastic: Error 400 (Bad Request): all shards failed [type=search_phase_execution_exception]

Status | 500
Status text | Internal Server Error
URL | /api/services
Response body | {   "data": null,   "total": 0,   "limit": 0,   "offset": 0,   "errors": [     {       "code": 500,       "msg": "Search service failed: elastic: Error 400 (Bad Request): all shards failed [type=search_phase_execution_exception]"     }   ] }
```

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #374 
Resolves #622 

Instead of creating indices with mapping the collector at start time creates index template. ES uses this index template when a data is inserted to ES.

TODO

- [x] a flag to disable template creation

### CODE REVIEW

**Comment by pavolloffay** [comment]
Please do not merge yet!

**Comment by pavolloffay** [comment]
cc @jaegertracing/elasticsearch this PR changes behavior if the writer. Now jaeger will not be creating indices, but instead it will create index template at startup. There will be also a flag to disable template creation if it was previously manually installed by a user - needed if jaeger does not have manage permissions on the cluster see https://github.com/jaegertracing/jaeger/issues/1691#issuecomment-515989052. 

Please comment if you have any concerns.

File: plugin/storage/es/options.go [hunk 1]
```diff
@@ -214,6 +216,10 @@ func addFlags(flagSet *flag.FlagSet, nsConfig *namespaceConfig) {
 		"(experimental) Use read and write aliases for indices. Use this option with Elasticsearch rollover "+
 			"API. It requires an external component to create aliases before startup and then performing its management. "+
 			"Note that "+nsConfig.namespace+suffixMaxSpanAge+" is not taken into the account and has to be substituted by external component managing read alias.")
+	flagSet.Bool(
```
Comment (pavolloffay) [inline plugin/storage/es/options.go]: @yardbirdsax I have added boolean flag `--es.create-index-templates`  which creates templates at startup when enabled. By default it is enabled.

**Comment by pavolloffay** [comment]
I will merge this by Friday. If you have any concerns please comment by then.

File: plugin/storage/es/options.go (hunk 1 above)
Comment (yardbirdsax) [reply plugin/storage/es/options.go]: That's awesome @pavolloffay , thanks!

**Comment by dzmitry-lahoda** [comment]
is it in release 1.14?

**Comment by pavolloffay** [comment]
Based on the changelog yes https://github.com/jaegertracing/jaeger/blob/master/CHANGELOG.md#1140-2019-09-02

**Pull request merged**